src/meta/processors/job/MetaJobExecutor.cpp

```diff
@@ -79,15 +79,21 @@ ErrOrHosts MetaJobExecutor::getTargetHost(GraphSpaceID spaceId) {
   }
 
   // use vector instead of set because this can convient for next step
+  std::unordered_map<HostAddr, std::vector<PartitionID>> hostAndPart;
   std::vector<std::pair<HostAddr, std::vector<PartitionID>>> hosts;
   while (iter->valid()) {
+    auto part = MetaServiceUtils::parsePartKeyPartId(iter->key());
     auto targets = MetaServiceUtils::parsePartVal(iter->val());
     for (auto& target : targets) {
+      hostAndPart[target].emplace_back(part);
       std::vector<PartitionID> parts;
       hosts.emplace_back(std::make_pair(std::move(target), std::move(parts)));
     }
     iter->next();
   }
+  for (auto it = hostAndPart.begin(); it != hostAndPart.end(); it++) {
```

Comment:
line 89 and 90 , should we delete these two lines?

Reply:
it seems useless .